app/Http/Controllers/WikiController.php

```diff
@@ -80,13 +80,13 @@ public function create(Request $request): \Illuminate\Http\Response
                 'value' => Str::random(64),
             ]);
 
-            // Enable elasticsearch for new wikis by default
+            // Create elasticsearch setting to be updated once ElasticSearch has been initialized
             // T285541
-//             WikiSetting::create([
-//                 'wiki_id' => $wiki->id,
-//                 'name' => 'wwExtEnableElasticSearch',
-//                 'value' => true,
-//             ]);
+            WikiSetting::create([
+                'wiki_id' => $wiki->id,
+                'name' => 'wwExtEnableElasticSearch',
+                'value' => false,
+            ]);
```

Comment:
If we wanted to ensure that no pages get created before ElasticSearch is ready (because we don’t want any content to bypass the index), we could set some internal setting that is then assigned to [`$wgReadOnly`](https://www.mediawiki.org/wiki/Manual:$wgReadOnly), so that the wiki only becomes writable once the search index is ready (i.e. change that setting in the init job as well). But right now we probably don’t need that.

Reply:
Indeed, and this may be something that we want to think about.
We could also force indexing of the whole site after enabling wwExtEnableElasticSearch / running the job to create the indexes?

Reply:
In the current case it wouldn't make much sense since the site will probably be pretty empty?

When this setting would be enabled for something that already has a lot of data it would make more sense to me to run the ForceSearchIndex thing